Add CHANGELOG containing release notes (#375)
Materials-Consortia/OPTIMADE: CHANGELOG.md

# Changelog

> ## v1.1.0 (08/07/2021)

Reviewer: Date formats is a curious thing, and one persons standard date format is another persons indecipherable scribbles. How do I know if this represents 7:th of August or 8:th of July? I usually suggest the one and only ISO format (see https://xkcd.com/1179/), but with time I've become more humble (eh...) and I am going to suggest that we adopt what I think is the least controversial format in this context:
```suggestion
## v1.1.0 (July 8, 2021)
```

This is release v1.1.0 of the OPTIMADE API specification.

> This is a minor release that primarily clarifies the specification, with only two additions.

Reviewer: (Due to changes suggested below)
```suggestion
This is a minor release that primarily patches minor specification errors and introduces one new feature.
```

> ### Changes
> 
> - The `mass` field of the `species` attribute for the `structures` entry type has been updated from a float to a list of floats ([#344](https://github.com/Materials-Consortia/OPTIMADE/pull/344)). This is a breaking change for any implementations that serve this optional field.
> - The `implementation` field of the general `meta` response has been updated to include an `issue_tracker` field ([#339](https://github.com/Materials-Consortia/OPTIMADE/pull/339)).
> 
> ### Fixes
> 
> - The specification text has been clarified in several places.
> - Multiple typos, grammatical errors, and incorrect API examples have been fixed.
> - The OpenAPI schemas are now fully compliant with the Swagger validator.

Reviewer: Is the distinction between *Changes* and *Fixes* clear? Since this changelog entry is likely going to be the format we copy+paste going forward, perhaps we should divide it into the semver categories? So, how about:

```suggestion
### New features

- The `implementation` field of the general `meta` response has been updated to include an `issue_tracker` field ([#339](https://github.com/Materials-Consortia/OPTIMADE/pull/339)).

### Patches

- The `mass` field of the `species` attribute for the `structures` entry type has been updated from a float to a list of floats ([#344](https://github.com/Materials-Consortia/OPTIMADE/pull/344)). The was deemed a specification bug that now is fixed in both the specification text and the schemas. (Note: software implemented to strictly adhere to the specification bug could break by this change.)
- The specification text has been clarified in several places without change of intended meaning.
- Multiple typos, grammatical errors, and incorrect API examples have been fixed.
- The OpenAPI schemas are now fully compliant with the Swagger validator.
```

Author: I prefer your suggestion, I made a couple of edits to your suggestion to fix typos and split the species.mass point into a few bulletpoints.

> ## v1.0.0 (01/07/2020)

Reviewer: ```suggestion
## v1.0.0 (July 1, 2020)
```

This is release v1.0.0 of the OPTIMADE API specification.

> The specification has undergone a few changes since v1.0.0-rc.2, some not backward compatible.
> 
> A few highlighted changes directly affecting the API:
> - Several changes related to version negotiation for future releases of the API have been added.
> - The API may now be served on both the unversioned and versioned URLs.
> - The top-level `meta` field has undergone major revision:
>    - It is now mandatory, with a few mandatory subfields (e.g., `api_version`), and most other fields made optional.
>    - A link to a schema may be provided.
>    - It no longer contains an `index_base_url`, since this functionality is covered by `root` links in the `/links` endpoint.
> 
> - The `unit` field for describing properties in the `/info` endpoints is now standardized to use the Unified Code for Units of Measure.

Reviewer: I'd argue just remove this and simply start with the "This is release v1.0.0 of the OPTIMADE API specification" for the first release. Going forward, it doesn't seem particular important to start the file with what was changed since specifically the v1.0.0-rc.2 prerelease. Isn't the most sensible place to formally start logging changes more carefully the v1.0.0 version?
```suggestion
```

Author: I was toying with this too, I'll remove it but I think it makes sense to leave this in the actual GitHub release notes.

Reviewer 2: I agree with this sentiment.
Hopefully no one has implemented versions prior to v1.0.0 at this stage - at least not in production.
On the other hand, it's neat to have this information somewhere for our sake as the developers.
